components/builder-api/src/server/resources/origins.rs

```diff
@@ -1101,6 +1103,49 @@ fn transfer_origin_ownership(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn depart_from_origin(req: HttpRequest, path: Path<String>, state: Data<AppState>) -> HttpResponse {
+    let origin = path.into_inner();
+
+    let session = match authorize_session(&req, None) {
+        Ok(session) => session,
+        Err(err) => return err.into(),
+    };
+
+    // Do not allow an origin owner to depart which would orphan the origin
+    if check_origin_owner(&req, session.get_id(), &origin).unwrap_or(false) {
+        return HttpResponse::new(StatusCode::FORBIDDEN);
+    }
+
+    // Pass a meaningful error in the case that the user isn't a member of origin
+    if !check_origin_member(&req, &origin, session.get_id()).unwrap_or(false) {
+        return HttpResponse::new(StatusCode::UNPROCESSABLE_ENTITY);
+    }
+
+    let conn = match state.db.get_conn().map_err(Error::DbError) {
+        Ok(conn_ref) => conn_ref,
+        Err(err) => return err.into(),
+    };
+
+    debug!(" Departing from origin {}", &origin);
+
+    let account_id = match Account::get(&session.get_name(), &*conn).map_err(Error::DieselError) {
```

Comment:
I believe you don't need to do this, the session id is the account id.. so you should be able to get it via `session.get_id()`